stdlib/2and3/xml/etree/ElementTree.pyi

```diff
@@ -95,20 +122,31 @@ if sys.version_info >= (3, 4):
         def close(self) -> None: ...
         def read_events(self) -> Iterator[Tuple[str, Element]]: ...
 
-def XML(text: AnyStr, parser: 'XMLParser'=...) -> Element: ...
-def XMLID(text: AnyStr, parser: 'XMLParser'=...) -> Tuple[Element, Dict[str, Element]]: ...
+def XML(text: _parser_input_type, parser: 'XMLParser'=...) -> Element: ...
+def XMLID(text: _parser_input_type, parser: 'XMLParser'=...) -> Tuple[Element, Dict[_str_result_type, Element]]: ...
 
-# TODO-improve this type
-fromstring = ...  # type: Callable[..., Element]
+# This is aliased to XML in the source.
+fromstring = XML
 
-def fromstringlist(sequence: Sequence[AnyStr], parser: 'XMLParser'=...) -> Element: ...
+def fromstringlist(sequence: Sequence[_parser_input_type], parser: 'XMLParser'=...) -> Element: ...
+
+# This type is both not precise enough and too precise. The TreeBuilder
+# requires the elementfactory to accept tag and attrs in its args and produce
+# some kind of object that has .text and .tail properties.
+# I've chosen to constrain the ElementFactory to always produce an Element
+# because that is how almost everyone will use it.
+# Unfortunately, the type of the factory arguments is dependent on how
+# TreeBuilder is called by client code (they could pass strs, bytes or whatever);
+# but we don't want to use a too-broad type, or it would be too hard to write
+# elementfactories.
+_ElementFactory = Callable[[Any, Any], Element]
```

Comment:
I noticed I was confused about variance in callable arguments. So I tried it and it worked with the Element constructor and with a hand-written function accepting Dict as well as Mapping. So I think you are right -- changed!